arches/app/media/js/viewmodels/workflow.js

```diff
@@ -50,48 +50,46 @@ define([
         });
         
         this.initialize = function() {
-            /* BEGIN workflow metadata logic */ 
             if (self.componentName) {
```

Comment:
overall lgtm! One minor change that isn't related to what you wrote.

Now that the component 100% depends on `self.componentName` , having that if-statement could be problematic for bugchasing in the future. Mind removing it?